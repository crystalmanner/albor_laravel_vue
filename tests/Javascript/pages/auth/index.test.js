import { shallowMount } from '@vue/test-utils'
import createStore from 'fresh-bus/store'
import Component from '~/pages/auth/index'

describe('Auth Page', () => {
  let localVue

  describe('Visuals', () => {
    test('hide register link and show title, logo and bg', async () => {
      const redirect = jest.fn()
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store: createStore({
          registrationType: 2,
          navigationAdmin: {
            logo: 'arbor-logo.png',
            background: 'arbor-background.png',
            title: 'Arbor Title'
          }
        }),
        mocks: {
          $auth: { redirect }
        }
      })
      expect(wrapper.html()).not.toContain("Don't have an account? Request access.")
      expect(wrapper.html()).toContain('Arbor Title')
      expect(wrapper.find('.background-image').attributes('style')).toContain('arbor-background.png')
      expect(wrapper.find('v-img-stub').attributes('src')).toContain('arbor-logo.png')
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
