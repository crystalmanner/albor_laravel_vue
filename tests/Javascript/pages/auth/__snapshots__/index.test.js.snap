// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Page Visuals hide register link and show title, logo and bg 1`] = `
<v-content-stub
  class="background-image"
  style="background-image: url(arbor-background.png);"
  tag="main"
>
  <v-container-stub
    fill-height=""
    fluid=""
    tag="div"
  >
    <v-layout-stub
      align-center=""
      justify-center=""
      tag="div"
    >
      <v-flex-stub
        md4=""
        sm8=""
        tag="div"
        xs12=""
      >
        <v-form-stub>
          <v-card-stub
            class="elevation-12 rounded-card"
            tag="div"
          >
            <v-toolbar-stub
              color="transparent"
              flat="true"
              scrollthreshold="300"
            >
              <v-img-stub
                contain="true"
                height="64"
                position="center center"
                src="arbor-logo.png"
                transition="fade-transition"
              />
            </v-toolbar-stub>
             
            <v-card-title-stub
              class="justify-center"
              primarytitle="true"
            >
              <div>
                <h3
                  class="headline mb-0"
                  style="justify-content: center;"
                >
                  
                  Log in to Arbor Title
                
                </h3>
              </div>
            </v-card-title-stub>
             
            <v-card-text-stub>
              <v-text-field-stub
                name="login"
                placeholder="Login"
                prepend-icon="person"
                type="text"
                value=""
              />
               
              <v-text-field-stub
                id="password"
                name="password"
                placeholder="Password"
                prepend-icon="lock"
                type="password"
                value=""
              />
            </v-card-text-stub>
             
            <div>
              <v-card-actions-stub>
                <v-btn-stub
                  activeclass="v-btn--active"
                  color="secondary"
                  flat="true"
                  small="true"
                  tag="button"
                  type="button"
                >
                  
                  Forgot password?
                
                </v-btn-stub>
                 
                <v-spacer-stub />
                 
                <v-btn-stub
                  activeclass="v-btn--active"
                  color="primary"
                  tag="button"
                  type="submit"
                >
                  
                  Login
                
                </v-btn-stub>
              </v-card-actions-stub>
               
              <!---->
            </div>
             
            <!---->
             
            <v-snackbar-stub
              color="error"
              timeout="6000"
              top="true"
            >
              
              
              
              <v-btn-stub
                activeclass="v-btn--active"
                dark="true"
                flat="true"
                tag="button"
                type="button"
              >
                
                Close
              
              </v-btn-stub>
            </v-snackbar-stub>
          </v-card-stub>
        </v-form-stub>
      </v-flex-stub>
    </v-layout-stub>
  </v-container-stub>
   
  <v-dialog-stub
    contentclass=""
    maxwidth="500"
    origin="center center"
    transition="dialog-transition"
    width="auto"
  >
    <forgot-password-stub />
  </v-dialog-stub>
</v-content-stub>
`;
